applications/vessel-history/src/features/profile/components/activity/description/ActivityDescription.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { Fragment } from 'react'
+import { ActivityEvent, ActivityEventGroup } from 'types/activity'
+import ActivityDescriptionEEZ from './ActivityDescriptionEEZ'
+
+interface ActivityDescriptionProps {
+  event?: ActivityEvent
+  group?: ActivityEventGroup
+}
+
+const ActivityDescription: React.FC<ActivityDescriptionProps> = (props): React.ReactElement => {
+  const event = props.event
+  const group = props.group
+
+  if (group?.event_eez){
+    return <ActivityDescriptionEEZ regionId={group?.event_eez} type="group" ocean={group.ocean}></ActivityDescriptionEEZ>
+  }
+
+  if (event?.regions.eez[0]){
+    return <ActivityDescriptionEEZ regionId={event.regions.eez[0]} type="event" ocean={event.regions.ocean[0]}></ActivityDescriptionEEZ>
+  }
+
+  if (group?.event_rfmo){
+    return <ActivityDescriptionEEZ regionId={group?.event_rfmo} type="group" ocean={group.ocean}></ActivityDescriptionEEZ>
+  }
+
+  if (event?.regions.rfmo[0]){
+    return <ActivityDescriptionEEZ regionId={event.regions.rfmo[0]} type="event" ocean={event.regions.ocean[0]}></ActivityDescriptionEEZ>
+  }
```

Comment:
There are too many `if` conditions in this component, this could probably be symptoms that something should be improved in the way we are structuring the data or the purpose/responsibility of this component.

Reply:
well this component is the one that is doing and improvement, verifing what data we have and calling the right descriptions so the selectors with ID are called only when the right id is present and is not possible to have the selector in a IF